### PR TITLE
chore: speed up deploy pipeline

### PR DESCRIPTION
**Pull request by frontier159**
# Description
What does this PR solve?

Fixes # (issue)

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

**Comment by nichosystem** [comment]
Deploy preview for _core_ ready!

✅ Preview
https://core-6fg9s5mg8-templedao.vercel.app
https://pr-977-preview.templedao.link

Built with commit c00c51fbb5b42656354ae515972463fd6e2a4c52.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

File: apps/dapp/vercel.json [hunk 1]
```diff
@@ -4,5 +4,6 @@
     "silent": false,
     "enabled": false,
     "autoJobCancelation": true
-  }
+  },
+  "buildCommand": "yarn vite:build"
```
Comment (marshall2112) [inline apps/dapp/vercel.json:8]: So now we are relying on the `prepare` step executing (more "magic") to generate the typechain artifacts. I'd personally prefer to be explicit. In other words, rename prepare to something like "gen-typechain" or whatever, and use `yarn build` as the build command (updated as `yarn gen-typechain && tsc && vite build`). Wdyt? Not a deal breaker, just curious if you agree. (On the other hand, we get the typechain for free today after yarn install, it probably improves the dev experience a bit...)

Comment (frontier159) [reply apps/dapp/vercel.json:8]: It always was relying on `prepare` to compile and get the typechain...but it then calling this multiple times.

If you bake into `yarn build` then it's going to try and compile and copy files every singe time you want to build or `yarn dev` or whatever. The contracts won't change very frequently, so think it's good enough to:
1. Build when yarn is first setup.
2. Then call explicitly on demand if/as you need it.

Comment (marshall2112) [reply apps/dapp/vercel.json:8]: I was thinking to only have the typechain compilation happening in certain steps, explicitly. Still, I'm fine to keep it as-is.

**Review by marshall2112 — APPROVED** [review]

**Pull request merged**